db/migrate/20170927135007_migrate_dialog_field_associations_to_use_new_relationship.rb

```diff
@@ -0,0 +1,31 @@
+class MigrateDialogFieldAssociationsToUseNewRelationship < ActiveRecord::Migration[5.0]
+  Rails.application.eager_load!
+
+  def up
+    Dialog.all.each do |dialog|
+      fields = dialog.dialog_fields
+      dialog_fields_with_associations = fields.select { |df| !df.auto_refresh.nil? || !df.trigger_auto_refresh.nil? }.sort { |n| n.position * -1 }
+      triggers = dialog_fields_with_associations.select { |df| df.trigger_auto_refresh == true }
```

Comment:
if trigger_auto_refresh is a boolean column, this simplifies to 

`.select(&:trigger_auto_refresh)`

(same with the next line)